src/main/java/dao/InvoiceDao.kt

```diff
@@ -45,11 +45,10 @@ class InvoiceDao {
             .setParameter("state", state)
             .resultList
 
-    fun getAllInvoicesByProfile(profileId: Long?): List<Invoice> {
-        val query = em!!.createQuery("SELECT i FROM Invoice i " + "JOIN i.profile c WHERE c.id = :id", Invoice::class.java)
-
-        return query.setParameter("id", profileId).resultList
-    }
+    fun allInvoicesByCivilian(civilianId: String): List<Invoice> = em
+            .createQuery("SELECT i FROM Invoice i JOIN i.profile p WHERE p.id = :id", Invoice::class.java)
```

Comment:
Aren't invoices linked to a profile? That way you should be able to get them without using a query

Reply:
Function has been removed, invoices linked by a profile now are received by the userDao